File: WordloomFrontend/next/src/modules/orbit/domain/notes.ts
export type Tag = {
  id: string;
  name: string;
  color: string;
  icon?: string | null;
  description?: string | null;
  count: number;
  createdAt?: string | null;
  updatedAt?: string | null;
};

export type Note = {
  id: string;
  title: string | null;
  summary?: string | null; // 新增：Note 摘要/描述
  summary_zh?: string | null; // 后端返回的摘要（蛇形命名）
  text: string;
  blocksJson?: string | null; // 新增：JSON格式的blocks存储
  status: string;
  priority: number; // 重要程度 1-5
  urgency?: number; // 紧急程度 1-5
  usageLevel?: number; // 日用程度 1-5
  usageCount?: number; // 使用次数（自动计数）
  tags: string[]; // 标签名称数组（保留向后兼容）
  tagsRel?: Tag[]; // 新的标签对象数组（包含颜色、描述等）
  tags_rel?: Tag[]; // 后端返回的标签对象数组（蛇形命名）
  bookshelfId?: string | null; // 所属 Bookshelf ID（可选）
  bookshelf_id?: string | null; // 后端返回的 Bookshelf ID（蛇形命名）
  dueAt?: string | null;
  createdAt?: string | null;
  updatedAt?: string | null;
  storagePath?: string; // 固定存储路径
  storage_path?: string; // 后端返回的存储路径（蛇形命名）
  previewImage?: string; // 首张图片用于预览
  preview_image?: string; // 后端返回的预览图片（蛇形命名）
  previewText?: string; // 预览文本
  preview_text?: string; // 后端返回的预览文本（蛇形命名）
  bookshelfTag?: { id: string; name: string; color?: string; icon?: string } | null; // 书架标签
  bookshelf_tag?: { id: string; name: string; color?: string; icon?: string } | null; // 后端返回的书架标签（蛇形命名）
  isPinned?: boolean; // 是否置顶
  is_pinned?: boolean; // 后端返回的置顶标记（蛇形命名）
  pinnedAt?: string | null; // 置顶时间
  pinned_at?: string | null; // 后端返回的置顶时间（蛇形命名）
  cover_image_url?: string | null; // 封面图 URL
  cover_image_display_width?: number; // 封面图显示宽度
};

export type NoteListParams = {
  q?: string;
  tag?: string;
  status?: string;
  sort?: string;
  limit?: number;
  offset?: number;
};

File: WordloomBackend/api/app/models/orbit/notes.py
from __future__ import annotations
from typing import TYPE_CHECKING
from sqlalchemy import Column, Text, Integer, DateTime, func, text, ForeignKey, Boolean
from sqlalchemy.dialects.postgresql import UUID, ARRAY
from sqlalchemy.orm import relationship
from app.database_orbit import OrbitBase

if TYPE_CHECKING:
    from app.models.orbit.bookshelves import OrbitBookshelf

class OrbitNote(OrbitBase):
    __tablename__ = "orbit_notes"

    id = Column(UUID(as_uuid=False), primary_key=True, server_default=text("gen_random_uuid()"))
    title = Column(Text, nullable=True)
    summary = Column(Text, nullable=True, default="")  # 新增：Note 摘要/描述
    content_md = Column(Text, nullable=True, default="")
    blocks_json = Column(Text, nullable=True, default="[]")  # 新增：JSON格式的blocks数组
    preview_image = Column(Text, nullable=True, default=None)  # 新增：封面图 URL
    status = Column(Text, nullable=False, server_default=text("'open'"))
    priority = Column(Integer, nullable=False, server_default=text("3"))
    # 紧急程度 1-5
    urgency = Column(Integer, nullable=False, server_default=text("'3'"))
    usage_level = Column(Integer, nullable=False, server_default=text("'3'"))
    # 使用次数（自动计数，用户点击时递增）
    usage_count = Column(Integer, nullable=False, server_default=text("0"))
    tags = Column(ARRAY(Text), nullable=False, server_default=text("'{}'::text[]"))
    due_at = Column(DateTime(timezone=True), nullable=True)
    is_pinned = Column(Boolean, nullable=False, server_default=text("false"))  # 置顶标记
    pinned_at = Column(DateTime(timezone=True), nullable=True)  # 置顶时间

    # 业界标准：存储路径固定，不随 bookshelf 改变
    # 格式：relative/path/notes/{note_id}
    storage_path = Column(Text, nullable=False, unique=True, index=True)

    created_at = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
    updated_at = Column(DateTime(timezone=True), nullable=False, server_default=func.now(), onupdate=func.now())

    # Bookshelf 关联（外键）- 只记录业务关系，不影响存储路径
    bookshelf_id = Column(UUID(as_uuid=False), ForeignKey("orbit_bookshelves.id", ondelete="SET NULL"), nullable=True, index=True)

    # 关系映射到新的标签系统
    tags_rel = relationship(
        "OrbitTag",
        secondary="orbit_note_tags",
        back_populates="notes"
    )

    # 关系映射到 Bookshelf（延迟加载，避免循环导入）
    bookshelf = relationship(
        "OrbitBookshelf",
        back_populates="notes",
        foreign_keys=[bookshelf_id],
        lazy="noload"  # 不自动加载，需要时显式查询
    )